added board_types.txt to hold all BOARD_TYPE values

tridge opened the pull request:
this is needed as the ArduPilot project has another implementation of
this bootloader protocol, and thus another source of BOARD_TYPE
values. To maintain a common list without collisions this file can be
used

dagar commented:
How would this file be kept in sync? At a minimum let's document something in the file itself.

tridge commented:
this will need to be kept in sync by having edits to this file when a new target is added.
I've added a comment to that effect in this file

davids5 commented:
I will look at this is detail ASAP

davids5 commented on board_types.txt:
```diff
@@ -0,0 +1,33 @@
+TARGET_HW_CUBE_F7                      50
+
+
+# values below this line are implemented in the ArduPilot bootloader
```
@tridge 

So that we can keep these in sync better. Can we point to the repo where the AP boot loaded in located in this text file?  Also I have a set to code points that need to be added as well. I would also asks that we keep the TARGET_HW prefix for inter op.

davids5 commented:
@ksschwabe 

 I think this is more a reservation system. It would be the Pixhawk Board ID Registry. Having the list sorted in numerical order would be the way to audit it. The indirection has no advantage. We do not want to change the ID by edition this file.

davids5 commented:
@tridge Here is what I have as used. There are some WIP boards that we should reserve the # as well. But I think we do not have any collisions. 

Boot Loader | Image |   | Build root name
-- | -- | -- | --
BOARD_TYPE | board_id | TARGET_HW_xxxx |  
5 | 5 | PX4_FMU_V1 | px4fmu-v1
6 | 6 | PX4_FLOW_V1 | N/A
10 | 7 | PX4_PIO_V1 | px4io-v1
9 | 9 | PX4_FMU_V2 | px4fmu-v2
10 | 10 | PX4_PIO_V2 | px4io-v2
11 | 11 | PX4_FMU_V4 | px4fmu-v4
13 | 13 | PX4FMU_V4PRO | px4fmu-v4pro
50 | 50 | PX4_FMU_V5 | px4fmu-v5
12 | 12 | CRAZYFLIE | crazyflie
98 | 19 | PX4_AEROCORE_V1 | aerocore
N/A | 22 | NA | px4cannode-v1
N/A | 23 | N/A | s2740vc-v1
N/A | 24 | N/A | px4flow-v2
N/A | 25 | N/A | px4esc-v1
N/A | 26 | N/A | zubaxgnss-v1
N/A | 27 | N/A | esc35-v1
WIP (28) | 28 | NXPHLITE_V1 | nxphlite-v1
WIP (28) | 29 | NXPHLITE_V3 | nxphlite-v3
33 | 33 | AUAV_X21 | auav_x21
64 | 64 | TAP_V1 | tap-v1
66 | 66 | WIP_66 | tbd
65 | 65 | AEROFC_V1 | aerofc-v1
67 | 67 | WIP_67 | N/A
88 | 88 | MINDPX_V2 | mindpx-v2
N/A | 90 | N/A | px4nucleoF767ZI-v1
  | 91 | WIP_91 | tbd
96 | 96 | WIP_96 | tbd
98 | 98 | PX4_AEROCORE_V2 | aerocore2_default
99 | 99 | PX4_DISCOVERY_V1 | px4-stm32f4discovery
110 | 110 | PX4_SAME70XPLAINED_V1 | px4-same70xplained-v1_default

davids5 commented:
@tridge - no we do on 50!

tridge commented:
@davids5 I've added links to the ArduPilot bootloader and hwdef files.
The dupe for 50 can be fixed - CUBE_F7 now uses the ArduPilot bootloader as "CUBE_ORANGE" with ID 120
I've added a commit to remove CUBE_F7

tridge commented:
I also added the RadioLink Mini-Pix. They are shipping with board type 3. I have no idea if they have released source for the bootloader change (probably not), but best to reserve the ID so someone else doesn't collide

tridge commented:
@davids5 ping?

tridge commented:
rebased on master

davids5 commented:
@tridge - can we have a call to discuss the evolution of the bootloader? Please ping me on skype

tridge commented:
@davids5 I've pinged you, although I think this PR should really go in now, or we're just going to get inevitable ID conflicts

dagar commented:
If this is just a list to reserve IDs then why can't it be a single unified list that's sorted numerically? 

What about the inevitable duplicates? For example aren't TARGET_HW_OMNIBUSF4SD and AP_HW_OMNIBUSF4PRO actually the same thing?

This file could be markdown (like https://github.com/PX4/Bootloader/pull/104#issuecomment-399568340) and contain these extra details in comments.

tridge commented:
@dagar sorting numerically will make it harder to work out which bootloader is using the ID, which is important
The omnibus IDs are different as they do not use a compatible layout (different space reserved for flash storage of parameters, which means a fw built for one won't work for the other)
